synapse/lib/parser.py

```diff
@@ -132,6 +146,44 @@ def subquery(self, kids, meta):
 
         return kid
 
+    @lark.v_args(meta=True)
```

Comment:
these methods aren't needed. You only need to do this if you want to do something special. You just need to add entries to ruleClassMap instead.